ss58-js/mod.ts

```diff
@@ -0,0 +1,89 @@
+import { blake2b } from "../deps/hashes.ts";
+import * as base58 from "../deps/std/encoding/base58.ts";
+
+import { hex } from "../util/mod.ts";
+
+import {
+  ALLOWED_ADDRESS_LENGTHS,
+  ALLOWED_NETWORK_PREFIXES,
+  ALLOWED_PUBLIC_KEY_LENGTHS,
+  CHECKSUM_LENGTH,
+  SS58PRE,
+} from "./constants.ts";
+
+export const encode = (prefix: number, pubKey: string): string => {
+  const pubKeyBytes = hex.decode(pubKey);
+
+  if (!ALLOWED_PUBLIC_KEY_LENGTHS.includes(pubKeyBytes.length)) {
+    throw new Error("Invalid public key length");
+  }
+
+  if (!ALLOWED_NETWORK_PREFIXES.includes(prefix)) {
```

Comment:
Not loving this lookup. Should prefix validation be optional/pluggable? Perhaps there could be an optional 3rd param that accepts a list of (to-be-treated-as-valid) prefixes.

```diff
- (prefix: number, pubKey: Uint8Array): Uint8Array;
+ (prefix: number, pubKey: Uint8Array, validPrefixes?: number[]): Uint8Array;
```

Thoughts?

Reply:
Yeap, optional prefix validation sounds good